Core/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -30,5 +31,25 @@ public static string GetLeadingWhiteSpaces (this SyntaxNode node)
       var numberOfSpaces = node.GetLeadingTrivia().ToString().Count (c => c.ToString() == " ");
       return Strings.Space (numberOfSpaces);
     }
+
+    public static bool HasParent (this SyntaxNode node) => node.Parent != null;
+
+    public static bool IsChildOfLocalDeclarationStatement (this SyntaxNode node)
+    {
+      while (node.HasParent())
+      {
+        if (node.Parent.IsKind (SyntaxKind.LocalDeclarationStatement))
+        {
+          return true;
+        }
+
+        node = node.Parent!;
+      }
+
+      return false;
+    }
+
+    public static bool IsChildOfFieldDeclaration (this SyntaxNode node)
+      => !IsChildOfLocalDeclarationStatement (node);
```

Comment:
There are more node types that local decl & field decl, if a node is not a descendant of a local decl, it's not automatically a descandant of a field decl.